common/script/fns/randomDrop.js

```diff
@@ -3,80 +3,129 @@ import content from '../content/index';
 import i18n from '../i18n';
 import { daysSince } from '../cron';
 import { diminishingReturns } from '../statHelpers';
+import { predictableRandom } from './predictableRandomNamespaced';
+import randomVal from './randomVal';
 
 // Clone a drop object maintaining its functions so that we can change it without affecting the original item
 function cloneDropItem (drop) {
-  return _.cloneDeep(drop, function (val) {
+  return _.cloneDeep(drop, (val) => {
     return _.isFunction(val) ? val : undefined; // undefined will be handled by lodash
   });
 }
 
-module.exports = function randomDrop (user, modifiers, req) {
-  var acceptableDrops, base, base1, base2, chance, drop, dropK, dropMultiplier, name, name1, name2, quest, rarity, ref, ref1, ref2, ref3, task;
+module.exports = function randomDrop (user, modifiers, req = {}) {
+  let acceptableDrops;
+  let chance;
+  let drop;
+  let dropK;
+  let dropMultiplier;
+  let quest;
+  let rarity;
+  let task;
+
   task = modifiers.task;
-  chance = _.min([Math.abs(task.value - 21.27), 37.5]) / 150 + .02;
-  chance *= task.priority * (1 + (task.streak / 100 || 0)) * (1 + (user._statsComputed.per / 100)) * (1 + (user.contributor.level / 40 || 0)) * (1 + (user.achievements.rebirths / 20 || 0)) * (1 + (user.achievements.streak / 200 || 0)) * (user._tmp.crit || 1) * (1 + .5 * (_.reduce(task.checklist, (function(m, i) {
-    return m + (i.completed ? 1 : 0);
-  }), 0) || 0));
+
+  chance = _.min([Math.abs(task.value - 21.27), 37.5]) / 150 + 0.02;
+  chance *= task.priority *
+    (1 + (task.streak / 100 || 0)) *
+    (1 + user._statsComputed.per / 100) *
+    (1 + (user.contributor.level / 40 || 0)) *
+    (1 + (user.achievements.rebirths / 20 || 0)) *
+    (1 + (user.achievements.streak / 200 || 0)) *
+    (user._tmp.crit || 1) * (1 + 0.5 * (_.reduce(task.checklist, (m, i) => {
```

Comment:
here we should make so it works if task.checklist doesn't exists, maybe passing an empty array?